src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final double arrayVolumeAdd = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object [DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
 
+        if (size == elementData.length) {
+            grow(size);
+        }
+        elementData[size] = value;
+        size++;
+
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, 0, size);
+        if (size == elementData.length) {
+            grow(size);
+        }
```

Comment:
```suggestion
        if (size == elementData.length) {
            grow(size);
        }
       checkIndex(index);
```
now we can use the standard checker